server/start.go

```diff
@@ -88,9 +89,26 @@ const (
 	FlagMempoolMaxTxs = "mempool.max-txs"
 )
 
+// StartCmdOptions defines options that can be customized in `StartCmd`
+type StartCmdOptions struct {
+	AppCreator      types.AppCreator
+	DefaultNodeHome string
+	// DBOpener can be used to customize db opening, for example customize db options or support different db backends.
+	DBOpener func(rootDir string, backendType dbm.BackendType) (dbm.DB, error)
+	// PostSetup can be used to setup extra services under the same cancellable context,
+	// it's not called in stand-alone mode, only for in-process mode.
+	PostSetup func(svrCtx *Context, clientCtx client.Context, ctx context.Context, g *errgroup.Group) error
+	// AddFlags add custom flags to start cmd
+	AddFlags func(cmd *cobra.Command)
+}
```

Comment:
Amazing :pray: , very supportive of this. Thanks!

Though I think that this could maybe be improved with the StartCmd api being:
`AppCreator, DefaultNodeHom, StartCmdOptions` (and those two required fields not being in the same spot as remaining options)

Can we document the behavior of which of these are required / which aren't?

Reply:
If we make the argument `&StartCmdOptions` as well, then it can be a pretty trivial upgrade for folks, to just add a `nil` arg

Reply:
maybe a new API `StartCmdWithOptions`

Reply:
Why not instead of that new API variadic options? So it isn't API breaking and isn't as well another API.

Reply:
> Why not instead of that new API variadic options? So it isn't API breaking and isn't as well another API.

do you mean a list of option setters similar to the baseapp options?

Reply:
> > Why not instead of that new API variadic options? So it isn't API breaking and isn't as well another API.
> 
> 
> 
> do you mean a list of option setters similar to the baseapp options?

Yes, exactly!